minard/burst.py

```diff
@@ -0,0 +1,52 @@
+import couchdb
+from . import app
+
+def load_burst_runs(offset):
+    """
+    Returns a dictionary with the burst runs loaded from couchdb. The dummy itterator is used as key
+    to keep the ordering from the couchdb query. The content of the couchdb document is stored as values.
+    This loads ALL the documents in burst database, ordered by run_subrun_burst logic.
+    The logic to limit and split the results per page was implemented.
+    """
+    server = couchdb.Server("http://snoplus:"+app.config["COUCHDB_PASSWORD"]+"@"+app.config["COUCHDB_HOSTNAME"])
+    db = server["burst"]
+
+    i = 0 #the counter keeps the ordering correct
+    results = {}
+    limit = 10    #this changes the # of lines per page, can be modified when the DB gets big
+    offset = offset
+    page = (offset / limit)
+    skip = page * limit
+    all = db.view('_design/burst/_view/burst_by_run', descending=False, skip=skip)
+    total = all.total_rows
+    offset = all.offset
+    for row in db.view('_design/burst/_view/burst_by_run', descending=False, limit=limit, skip=skip):
+        run = row.key[0]
+        run_id = row.id
+        try:
+            results[i] = dict(db.get(run_id).items())
+        except KeyError:
+            app.logger.warning("Code returned KeyError searching for burst information in the couchDB. Run Number: %d" % run)
+        i += 1
```

Comment:
Still not sure I understand the logic here. Wouldn't this be equivalent to using a list, i.e.

```python
results = []
for row in db.view():
    results.append(dict(db.get(run_id).items()))
```

In both cases you can access the elements via `results[i]` right?

Reply:
The i is not there for the access but to keep the ordering (as is loaded from couchdb) correct.
The results come ordered from couch in a way (as run_subrun_burst because of the way the view is defined) that I want to preserve. 

I can't use the _id_ because those would get mixed while being passed from .py to the html page (I've tried this and they get out of order) - ie the runs would not be ordered by run_subrun_burst on the overview page anymore - and I can't use just the run number because there are several burst for the same run. I could keep the convention to use run_subrun_burst or I could just use the dummy i to keep the ordering the same as it arrives from couch. This was much simpler and does the same job.

Reply:
Ok, I think it's fine, but I'm not suggesting to use a different key, I'm suggesting to use a list which is naturally ordered from 1 to the number of elements.

Reply:
it definitely works in different ways, but I had issues with the data being out of order on the page when using different keys and this worked well.